### PR TITLE
feat: LiquidityManager and TokenManager contracts

### PR DESCRIPTION
**Pull request by RauppRafael**

### CODE REVIEW

File: contracts/libraries/InvestLib.sol [hunk 1]
```diff
@@ -118,33 +315,280 @@ library InvestLib {
...
+    function _collectPositionsLiquidity(
+        LiquidityPosition[] memory _positions
+    ) private returns (uint[][] memory) {
+        uint[][] memory withdrawnAmounts = new uint[][](_positions.length);
+
+        for (uint i; i < _positions.length; ++i) {
+            LiquidityPosition memory position = _positions[i];
+            INonfungiblePositionManager positionManager = INonfungiblePositionManager(position.positionManager);
+
+            (uint256 amount0, uint256 amount1) = positionManager.collect(
```
Comment (matheus-pissaia) [inline contracts/libraries/InvestLib.sol:562]:
```suggestion
            (uint amount0, uint amount1) = positionManager.collect(
```

File: contracts/StrategyManager.sol [hunk 2]
```diff
@@ -88,25 +95,35 @@ contract StrategyManager is HubOwnable, UseTreasury, ICall {
...
+    mapping(uint => InvestLib.TokenInvestment[]) internal _tokenInvestmentsPerStrategy;
 
     mapping(address => Position[]) internal _positions;
     // @dev investor => strategy position id => dca position ids
     mapping(address => mapping(uint => uint[])) internal _dcaPositionsPerPosition;
     // @dev investor => strategy position id => vault positions
     mapping(address => mapping(uint => InvestLib.VaultPosition[])) internal _vaultPositionsPerPosition;
+    // @dev investor => strategy position id => liquidity positions
+    mapping(address => mapping(uint => InvestLib.LiquidityPosition[])) internal _liquidityPositionsPerPosition;
+    // @dev investor => strategy position id => liquidity positions
```
Comment (matheus-pissaia) [inline contracts/StrategyManager.sol:116]:
```suggestion
    // @dev investor => strategy position id => token positions
```

File: contracts/StrategyManager.sol [hunk 3]
```diff
@@ -209,6 +258,21 @@ contract StrategyManager is HubOwnable, UseTreasury, ICall {
             _vaultInvestmentsPerStrategy[strategyId].push(vaultStrategy);
         }
 
+        for (uint i; i < _params.liquidityInvestments.length; ++i) {
+            InvestLib.LiquidityInvestment memory liquidityStrategy = _params.liquidityInvestments[i];
+
+            if (
+                !liquidityManager.positionManagerWhitelist(liquidityStrategy.positionManager) ||
+            liquidityStrategy.token0 > liquidityStrategy.token1
```
Comment (matheus-pissaia) [inline contracts/StrategyManager.sol:266]:
```suggestion
            liquidityStrategy.token0 >= liquidityStrategy.token1
```

File: hardhat.config.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ const config: HardhatUserConfig = {
                 settings: {
                     optimizer: {
                         enabled: true,
-                        runs: 1600,
+                        runs: 800,
```
[the remark this answers is not in the document]
Comment (RauppRafael) [reply hardhat.config.ts:32]: Because it reduces the contract's bytecode size

**Review by matheus-pissaia — APPROVED** [review]

**Pull request merged**